src/transforms/pipelines/config.rs

```diff
@@ -42,65 +52,167 @@ impl Clone for PipelineConfig {
     }
 }
 
+#[async_trait::async_trait]
+#[typetag::serde(name = "pipeline")]
+impl TransformConfig for PipelineConfig {
+    async fn build(&self, ctx: &TransformContext) -> crate::Result<Transform> {
+        let condition = if let Some(config) = &self.filter {
+            Some(config.build(&ctx.enrichment_tables)?)
+        } else {
+            None
+        };
+
+        // Setup the interior transform chain in this pipeline.
+        //
+        // NOTE: As this pipeline is not expanding to separate transforms we
+        // lose validation that happens at the topology layer (e.g. making sure
+        // input/output types of connected transforms are compatible). This is
+        // an unfortunate consequence of having a "topology in a topology".
+        //
+        // In the future, it would be great to decouple that type of validation
+        // from the physical task boundaries. You could imagine a future where
+        // we replace transform expansion with the ability to return different
+        // types of compound transforms directly from TransformConfig::build,
+        // and those know how to play nice with a subsequent graph-based
+        // validation scheme.
+        //
+        // For now, considering that most everything is log to log with regard
+        // to pipelines we avoid adding duplicate validation here in favor of
+        // future work.
+        let mut transforms = Vec::with_capacity(self.transforms.len());
+        for config in &self.transforms {
+            let transform = match config.build(ctx).await? {
+                Transform::Function(transform) => Box::new(transform),
+                Transform::Synchronous(transform) => transform,
+                _ => panic!("non-sync transform in pipeline: {:?}", config),
+            };
+            transforms.push(transform);
+        }
+
+        let buf_in = TransformOutputsBuf::new_with_capacity(
+            vec![Output::default(DataType::all())],
```

Comment:
Technically speaking, this needs the combined set of named outputs from every transform in the pipeline even though we ignore anything other than the default output. The only case I can think of where that could matter practically is if a remap transform tries to write to its error output. As is, there is no error output defined, which will cause a panic on `push_named`. Another option would be to give the output buffer an "ignore unknown" mode, since the nature of pipelines means don't care about anything but the primary.

Reply:
Ah ha, good point. 

TBH as a first impression defining an error output here makes more sense than silently dropping errors, especially if we aim to have pipelines be THE way to structure big configurations.

Reply:
True, that's another option. There's definitely a larger product discussion to be had around pipelines and multiple outputs, but I assume right now there _shouldn't_ be anything intentionally configured to produce to a non-default output.

Reply:
From a UX standpoint we're limited by what Datadog Pipelines offers, and they do not allow outputs from individual pipelines. The pipelines compound to produce a single output.